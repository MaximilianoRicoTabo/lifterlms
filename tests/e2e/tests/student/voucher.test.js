/**
 * Test voucher redemption on the student dashboard
 *
 * @since 4.12.0
 */

import {
	clickAndWait,
	createUser,
	createVoucher,
	fillField,
	loginStudent,
	logoutUser,
	visitPage,
} from '@lifterlms/llms-e2e-test-utils';

describe( 'StudentDashboard/RedeemVoucher', () => {

	afterEach( async () => {
		await logoutUser();
	} );

	it ( 'Should redeem a valid voucher', async () => {

		// Setup.
		const
			codes   = await createVoucher( { codes: 1, uses: 1 } ),
			student = await createUser();

		await logoutUser();

		// Use the voucher.
		await loginStudent( student.email, student.password );
		await visitPage( 'dashboard/redeem-voucher' );
		await fillField( '#llms-voucher-code', codes[0] );
		await clickAndWait( '#llms-redeem-voucher-submit' );

		// Success.
		expect( await page.$eval( '.llms-notice.llms-success', el => el.textContent ) ).toMatchSnapshot();

		await page.waitForSelector( '.llms-notification .llms-notification-main' );
		expect( await page.$eval( '.llms-notification .llms-notification-main', el => el.innerHTML ) ).toMatchSnapshot();

	} );

	it ( 'Should display an error for an invalid voucher', async() => {

		const student = await createUser();
		await logoutUser();

		await loginStudent( student.email, student.password );
		await visitPage( 'dashboard/redeem-voucher' );
		await fillField( '#llms-voucher-code', 'fakecode' );
		await clickAndWait( '#llms-redeem-voucher-submit' );

		// Error message.
		expect( await page.$eval( '.llms-notice.llms-error', el => el.textContent ) ).toMatchSnapshot();

	} );

} );
