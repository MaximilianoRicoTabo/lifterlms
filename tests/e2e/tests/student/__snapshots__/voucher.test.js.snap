// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentDashboard/RedeemVoucher Should display an error for an invalid voucher 1`] = `
"
			Voucher code \\"fakecode\\" could not be found.
	"
`;

exports[`StudentDashboard/RedeemVoucher Should redeem a valid voucher 1`] = `"Voucher redeemed successfully!"`;

exports[`StudentDashboard/RedeemVoucher Should redeem a valid voucher 2`] = `" <h4 class=\\"llms-notification-title\\">Course enrollment success!</h4> <div class=\\"llms-notification-body\\"><p>Congratulations! You enrolled in LifterLMS Quickstart Course</p> </div> "`;
